src/ripple/app/main/DBInit.cpp

```diff
@@ -90,14 +90,29 @@ const char* LedgerDBInit[] =
     );",
     "CREATE INDEX IF NOT EXISTS SeqLedger ON Ledgers(LedgerSeq);",
 
+    // InitialSeq field is the current ledger seq when the row
+    // is inserted. Only relevant during online delete
     "CREATE TABLE IF NOT EXISTS Validations   (                   \
+        LedgerSeq   BIGINT UNSIGNED,                \
+        InitialSeq  BIGINT UNSIGNED,                \
         LedgerHash  CHARACTER(64),                  \
         NodePubKey  CHARACTER(56),                  \
         SignTime    BIGINT UNSIGNED,                \
         RawData     BLOB                            \
     );",
+    // This will error out if the column already exists,
+    //  but DatabaseCon intentionally ignores errors.
+    "ALTER TABLE Validations                        \
+        ADD COLUMN LedgerSeq       BIGINT UNSIGNED;",
+    "ALTER TABLE Validations                        \
+        ADD COLUMN InitialSeq      BIGINT UNSIGNED;",
+
```

Comment:
I did a lot of digging into this, and as far as I can find, it's completely impossible with sqlite's SQL, and doing it in C++ would (a) require reimplementing this to use more complicated logic than just a array of strings, and (2) the implementation would be quite a bit more complicated than "ignore SQL errors".